### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by nalimilan**
Add documentation about indexing, improve various things. Replace `DataFrames.DataFrame` with `DataFrame`. Use `df.col` instead of `df[:col]` where possible.

### CODE REVIEW

File: docs/src/man/subsets.md [hunk 1]
```diff
@@ -1,109 +0,0 @@
-# Subsets
```
Comment (nalimilan) [inline docs/src/man/subsets.md]: This file was missing from the ToC. It's redundant with the new contents, except for `filter` and `filter!`, which don't really like advertising given that they are quite slow currently.

File: docs/src/man/getting_started.md [hunk 2]
```diff
@@ -227,30 +427,4 @@ df = DataFrame(x = 1, y = 2)
 CSV.write(output, df)
 ```
 
-The behavior of CSV functions can be adapted via keyword arguments. For more information, use the REPL [help-mode](http://docs.julialang.org/en/stable/manual/interacting-with-julia/#help-mode) or checkout the online [CSV.jl documentation](https://juliadata.github.io/CSV.jl/stable/).
-
-## Loading a Classic Data Set
```
Comment (nalimilan) [inline docs/src/man/getting_started.md]: I've removed this as it feels out of place (it's not used in this page). Other pages include code to load it anyway.

File: docs/src/man/getting_started.md [hunk 3]
```diff
@@ -118,68 +162,223 @@ but for very large `DataFrame`s  this may be a consideration.
...
+   9
+  11
+   ⋮
+ 991
+ 993
+ 995
+ 997
+ 999
+
+julia> df[:A] === df[:, :A]
```
Comment (bkamins) [inline docs/src/man/getting_started.md]: This is going to be deprecated (also above), only `==` will pass in the future as we want to be consistent with indexing in Base, where `:` creates a copy.

File: docs/src/man/joins.md [hunk 4]
```diff
@@ -6,14 +6,14 @@ We often need to combine two or more data sets together to provide a complete pi
 julia> using DataFrames
 
 julia> names = DataFrame(ID = [20, 40], Name = ["John Doe", "Jane Doe"])
-2×2 DataFrames.DataFrame
+2×2 DataFrame
```
Comment (bkamins) [inline docs/src/man/joins.md]: Here and below need to regenerate tables to include column eltype

File: docs/src/man/missing.md [hunk 5]
```diff
@@ -54,9 +52,13 @@ julia> collect(skipmissing(x))
 
 ```
 
-`missing` elements can be replaced with other values via `Missings.replace`.
+The [Missings.jl](https://github.com/JuliaData/Missings.jl) package provides a few convenience functions to work with missing values.
+
+The function `Missings.replace` replaces `missing` elements with another value:
```
Comment (bkamins) [inline docs/src/man/missing.md]: Why do we recommend it over `coalesce.(x,1)` (I know it is lazy, but is it that useful?)

File: docs/src/man/missing.md [hunk 6]
```diff
@@ -69,32 +71,34 @@ The function `Missings.T` returns the element-type `T` in `Union{T, Missing}`.
 
 ```jldoctest missings
 julia> eltype(x)
-Union{Int64, Missings.Missing}
+Union{Int64, Missing}
```
Comment (bkamins) [inline docs/src/man/missing.md]: And here the same question against `Base.nonmissingtype`?

File: docs/src/man/querying_frameworks.md [hunk 7]
```diff
@@ -15,8 +16,10 @@ A simple example of a query looks like this:
 ```jldoctest query
 julia> using DataFrames, Query
 
-julia> df = DataFrame(name=["John", "Sally", "Roger"], age=[54., 34., 79.], children=[0, 2, 4])
-3×3 DataFrames.DataFrame
+julia> df = DataFrame(name=["John", "Sally", "Roger"],
+                      age=[54., 34., 79.],
+                      children=[0, 2, 4])
+3×3 DataFrame
```
Comment (bkamins) [inline docs/src/man/querying_frameworks.md]: tables missing eltype information need regeneration below

File: docs/src/man/reshaping_and_pivoting.md [hunk 8]
```diff
@@ -8,7 +8,7 @@ julia> using DataFrames, CSV
 julia> iris = CSV.read(joinpath(dirname(pathof(DataFrames)), "../test/data/iris.csv"));
 
 julia> head(iris)
-6×5 DataFrames.DataFrame
+6×5 DataFrame
```
Comment (bkamins) [inline docs/src/man/reshaping_and_pivoting.md]: Same here with eltype and below

File: docs/src/man/sorting.md [hunk 9]
```diff
@@ -10,7 +10,7 @@ julia> iris = CSV.read(joinpath(dirname(pathof(DataFrames)), "../test/data/iris.
 julia> sort!(iris);
 
 julia> head(iris)
-6×5 DataFrames.DataFrame
+6×5 DataFrame
```
Comment (bkamins) [inline docs/src/man/sorting.md]: eltype problem with this file also

File: src/abstractdataframe/abstractdataframe.jl [hunk 10]
```diff
@@ -651,7 +651,7 @@ returns `true`. The function is passed a `DataFrameRow` as its only argument.
 # Examples
 ```
 julia> df = DataFrame(x = [3, 1, 2, 1], y = ["b", "c", "a", "b"])
-4×2 DataFrames.DataFrame
+4×2 DataFrame
```
Comment (bkamins) [inline src/abstractdataframe/abstractdataframe.jl]: add eltype information

File: docs/src/man/missing.md (hunk 5 above)
Comment (nalimilan) [reply docs/src/man/missing.md]: Good idea. I now present `coalesce` first.

File: docs/src/man/missing.md (hunk 6 above)
Comment (nalimilan) [reply docs/src/man/missing.md]: `Base.nonmissingtype` isn't part of the public API, and initially it has only been added for internal use. Hopefully we'll be able to support something like `typesubtract(T, Missing)` at some point. For now I prefer recommending `Missings.T`, which is public even if it's not exported.

**Comment by nalimilan** [comment]
Thanks. I've included changes in the commits. Regarding the eltypes, better fix that separately.

BTW, have you considered moving some of the material from Julia-DataFrames-Tutorial here?

File: docs/src/man/missing.md (hunk 6 above)
Comment (bkamins) [reply docs/src/man/missing.md]: OK

**Comment by bkamins** [comment]
OK - so let us merge it. I will update `eltypes` in a second run of clean up when I read all the material after merging on the website.

**Pull request merged**